src/patternfly/components/Table/table-grid.scss

```diff
@@ -177,6 +177,11 @@
     }
   }
 
+  // Remove border on tr inside non-expanded tbody in of expandable tables
+  &.pf-m-expandable tbody:not(.pf-m-expanded) tr {
+    border-block-end: 0;
+  }
+
```

Comment:
This does what we want, but something like this would be more straightforward. Disable all `<tr>` `border-block-end`. Initially, this logic applied to a mix and match approach to expandable table (tbody.expandable next to a regular row). However, [React's guidance](https://www.patternfly.org/components/table#expandable) places all expandable table content in a loop, within a `Tbody`. I believe it's safe to take this approach. 

**Add this variable to `:root` (`.#{$table}[class*="pf-m-grid"]` in this case) to avoid potential conflicts**

```
--#{$table}__tbody--responsive--m-expandable--BorderBlockEndWidth: var(--pf-t--global--border--width--divider--default);
```

```
  &.pf-m-expandable {
    --#{$table}__tr--BorderBlockEndWidth: 0; // nested table rows have no border

    .#{$table}__tbody {
      border-block-end: var(--#{$table}__tbody--responsive--border-width) solid var(--#{$table}--responsive--BorderColor);
    }
  }
```

You should then be able to remove [lines 190-199](https://github.com/patternfly/patternfly/pull/6698/files#diff-1c73508a6dac9de0a0052bcd77f9193e0f8d95a4d5cfef221569b6b97af1e649R190-R199) and [lines 201-212](https://github.com/patternfly/patternfly/pull/6698/files#diff-1c73508a6dac9de0a0052bcd77f9193e0f8d95a4d5cfef221569b6b97af1e649R201-R212), but because `pf-v-table` is very complex, we should run a visual regression test before merging.

Reply:
@mattnolting I've made these changes but noticed I also needed to specifically target `.#{$table}__tbody.pf-m-expanded` to apply as needed so added that specific rule to what you laid out above 👍